### ISSUE TITLE
Fix Stack Splitter for suite2p compatibility

### ISSUE THREAD

**Issue by cawarwick** [cawarwick/ThorStackSplitter#2]
We need to include a secondary condition in the stack splitter to make sure that whatever it splits it by, the resultant number of frames in each stack ALSO needs to be divisible by Z*C (in this case 10) with no remainder. I think the only thing I did was have it figure out how many pieces it needed to be in order to be less than 4gb and then check whether it was possible to split the file into that many even files (e.g. if it's not divisible by 3) and then increment it and check again.

psuedocode:

If # of splits needed to be <4gb is 3:
        Then find the number of frames in an individual split and make sure it's evenly divisible by C*Z
                  If there is a remainder:
                                Increment the number of splits and then check the condition
While (Total number/(C*Z)!=0, ++ stacks)

**Comment by cawarwick** [comment]
Or maybe use the substack function? Might be more straightforwward?

**Comment by cawarwick** [comment]
There were problems in 2 spots.
the first was that when removing NaNs from the recording stopping early it wouldn't care what the last number was e.g. 2447 is not a easily divisible number for splitting 2,3,4,5. The fix I implemented just rounds to the nearest 10, so some frames will be lost when there are NaNs present, but it will be at most 9 frames, e.g. ~1 second of recording.

I also implemented a double conditional during the stack splitting itself such that it checks if it can be split by that number and also whether each individual tif can be reassembled correctly by Suite2p/Imagej.
